sdk/deviceprovisioningservices/Microsoft.Azure.Management.DeviceProvisioningServices/tests/ScenarioTests/DeviceProvisioningClientCertificates.cs

```diff
@@ -1,55 +1,74 @@
 
 namespace DeviceProvisioningServices.Tests.ScenarioTests
 {
     public class DeviceProvisioningClientCertificates : DeviceProvisioningTestBase
     {
-
         [Fact]
-        public void CreateAndDelete()
+        public async Task CreateAndDelete()
         {
-            using (var context = MockContext.Start(this.GetType()))
-            {
-                this.Initialize(context);
-                var testName = "unitTestingDPSCertificatesCreateAndDelete";
-                var resourceGroup = this.GetResourceGroup(testName);
-                var service = this.GetService(resourceGroup.Name, testName);
-
-                //add a cert
-                this.provisioningClient.DpsCertificate.CreateOrUpdate(testName,
-                    testName, Constants.Certificate.Name,
-                    new CertificateBodyDescription(Constants.Certificate.Content));
-
-                var certificateList = this.provisioningClient.DpsCertificates.List(testName,
-                    testName);
-
-                Assert.Contains(certificateList.Value, x => x.Name == Constants.Certificate.Name);
-
-                //verify certificate details
-                var certificateDetails =
-                    certificateList.Value.FirstOrDefault(x => x.Name == Constants.Certificate.Name);
-                Assert.NotNull(certificateDetails);
-                Assert.Equal(certificateDetails.Properties.Subject, Constants.Certificate.Subject);
-                Assert.Equal(certificateDetails.Properties.Thumbprint, Constants.Certificate.Thumbprint);
-
-                //can get a verification code
-                var verificationCodeResponse =
-                    this.provisioningClient.DpsCertificate.GenerateVerificationCode(certificateDetails.Name,
-                        certificateDetails.Etag, resourceGroup.Name, service.Name);
-                Assert.NotNull(verificationCodeResponse.Properties);
-                Assert.False(string.IsNullOrEmpty(verificationCodeResponse.Properties.VerificationCode));
-
-                //delete certificate
-                this.provisioningClient.DpsCertificate.Delete(resourceGroup.Name, verificationCodeResponse.Etag, service.Name, Constants.Certificate.Name);
-                certificateList = this.provisioningClient.DpsCertificates.List(testName,
-                    testName);
-                Assert.DoesNotContain(certificateList.Value, x => x.Name == Constants.Certificate.Name);
-            }
+            using MockContext context = MockContext.Start(GetType());
+            Initialize(context);
+            var testName = "unitTestingDPSCertificatesCreateAndDelete";
+            ResourceGroup resourceGroup = await GetResourceGroupAsync(testName).ConfigureAwait(false);
+            ProvisioningServiceDescription service = await GetServiceAsync(resourceGroup.Name, testName).ConfigureAwait(false);
+
+            //add a cert
+            await _provisioningClient.DpsCertificate
+                .CreateOrUpdateAsync(
```

Comment:
Here's an example of one of the breaking changes. The certificate content is now a direct parameter instead of in a type `CertificateBodyDescription`.

Reply:
I am not sure. But this is most likely autorest bug fix where we no longer blindly double wrap a single content.

Reply:
If you're okay with that as a breaking change, and major rev of the version, then we're good.